src/utils/questionnaireBuilderUtils.ts

```diff
@@ -1,245 +1,331 @@
+ * @param index - The index of the question to be edited
+ * @param updatedQuestion - The new question data to replace the existing question
+ * @param questionnaire - The current questionnaire state object
+ * @returns Updated questionnaire object with the edited question
  */
 export const editQuestion = (
-  index: number, 
+  index: number,
   updatedQuestion: Question,
   questionnaire: Questionnaire
 ): Questionnaire => {
+  const updatedQuestions = [...(questionnaire.questions || [])];
+  updatedQuestions[index] = updatedQuestion;
+  
   return {
     ...questionnaire,
-    questions: questionnaire.questions.map((question, i) =>
-      i === index ? updatedQuestion : question
-    ),
+    questions: updatedQuestions
   };
 };
 
 /**
- * Function to count all correct answers in the questionnaire, used for passScore determination
+ * Function to count the total number of correct answers in a questionnaire
  * 
- * @param questionnaire Current questionnaire state
- * @returns Number of correct answers
+ * @param questionnaire - The questionnaire to count correct answers for
+ * @returns The total count of correct answers across all questions
  */
 export const countCorrectAnswers = (questionnaire: Questionnaire): number => {
-  return questionnaire.questions.reduce((count, question) => {
-    return (
-      count +
-      (question.answerOptions?.filter((option) => option.isCorrect).length ||
-        0)
-    );
+  if (!questionnaire.questions) return 0;
+  
+  return questionnaire.questions.reduce((total, question) => {
+    if (!question.answerOptions) return total;
+    
+    const correctAnswersCount = question.answerOptions.filter(
+      option => option.isCorrect
+    ).length;
+    
+    return total + correctAnswersCount;
   }, 0);
 };
 
 /**
- * Function to check and set the tooltip message based on form state
+ * Creates an empty questionnaire object with default values
  * 
- * @param questionnaire Current questionnaire state
- * @returns Tooltip message
+ * @returns A new empty questionnaire object
  */
-export const getTooltipMessage = (questionnaire: Questionnaire): string => {
-  if (!questionnaire.title || !questionnaire.description) {
-    return "Please provide both title and description";
-  }
+export const createEmptyQuestionnaire = (): Questionnaire => {
+  return {
+    title: "",
+    description: "",
+    questions: [],
+    passScore: 0,
+    tags: []
+  };
+};
 
+/**
+ * Validates a questionnaire and returns validation result with error message if invalid
+ * 
+ * @param questionnaire - The questionnaire to validate
+ * @returns An object containing isValid flag and an error message code if invalid
+ */
+export const validateQuestionnaire = (questionnaire: Questionnaire): { 
+  isValid: boolean; 
+  errorType: string | null;
+} => {
+  // Check for title and description
+  if (!questionnaire.title && !questionnaire.description) {
+    return { 
+      isValid: false, 
+      errorType: "bothEmpty" 
```

Comment:
Using errortype like this is not the best approach.
I think the best option in the switch is to have these conditions rather than the errorType string, that would then make this validateQuestionnaire function redundant, and so could be removed.
If you disagree lets discuss but I don't think this is necessary and it just adds another layer of parsing and complexity.
If for somereason this was the way to go I'd suggest using an enum rather than just a string. But for now I think this could be removed as suggested